src/flags.zig

```diff
@@ -851,7 +853,7 @@ test "flags" {
 
     try t.check(&.{ "pos", "x", "y" }, snap(@src(),
         \\stdout:
-        \\p1: y
+        \\p1: x
```

Comment:
We had this particular test since forever. This is the dark side of snaptests --- might be too easy to gloss over an incorrect result!

Reply:
But (all) fuzz tests caught us here :rofl: 

![image](https://github.com/tigerbeetle/tigerbeetle/assets/1711539/e8f4f544-c77b-4908-a8c2-aba5f4fbe5e0)